xds/build.gradle

```diff
@@ -42,3 +49,25 @@ if (rootProject.childProjects.containsKey('grpc-compiler')) {
 }
 
 configureProtoCompilation()
+
+jar {
+    classifier = 'original'
+}
+
+shadowJar {
+    classifier = null
+    dependencies {
+        include(project(':grpc-xds'))
+    }
+    relocate 'io.envoyproxy', 'io.grpc.xds.shaded.io.envoyproxy'
+    relocate 'validate', 'io.grpc.xds.shaded.com.lyft.pgv.validate'
+    exclude "**/*.proto"
+}
+
+publishing {
+    publications {
+        maven(MavenPublication) {
+            artifact shadowJar
```

Comment:
We'd normally also remove the original jar, like done for alts. Although I understand we aren't publishing for this project yet.